pkg/provider/api.go

```diff
@@ -425,6 +424,7 @@ func (ap *ApiProvider) RefreshUsers(ctx context.Context) error {
 			}
 			ap.logger.Info("Loaded users from Redis cache",
 				zap.String("team_id", teamID),
+				zap.String("user_id", userID),
 				zap.Int("count", len(cachedUsers)))
 			ap.usersReady = true
```

Comment:
Unrelated but can this be called concurrently? Consider making it atomic if yes. All good, if that's not the case.

Reply:
good catch fixed